nano/node/process_live_dispatcher.cpp

```diff
@@ -0,0 +1,59 @@
+#include <nano/lib/blocks.hpp>
+#include <nano/node/blockprocessor.hpp>
+#include <nano/node/election_scheduler.hpp>
+#include <nano/node/process_live_dispatcher.hpp>
+#include <nano/node/vote_cache.hpp>
+#include <nano/node/websocket.hpp>
+#include <nano/secure/common.hpp>
+#include <nano/secure/ledger.hpp>
+#include <nano/secure/store.hpp>
+
+nano::process_live_dispatcher::process_live_dispatcher (nano::ledger & ledger, nano::election_scheduler & scheduler, nano::vote_cache & inactive_vote_cache, nano::websocket_server & websocket) :
+	ledger{ ledger },
+	scheduler{ scheduler },
+	inactive_vote_cache{ inactive_vote_cache },
+	websocket{ websocket }
+{
+}
+
+void nano::process_live_dispatcher::connect (nano::block_processor & block_processor)
+{
+	block_processor.batch_processed.add ([this] (auto const & batch) {
+		auto const transaction = ledger.store.tx_begin_read ();
+		for (auto const & [result, block] : batch)
```

Comment:
One thing I didn't like is that each use of this observer is multiplier to the iteration list. Not every data is used, the size of the list could be a "target" set in a future improvement.